apollo-router/src/warp_http_server_factory.rs

```diff
@@ -302,22 +302,25 @@ where
         Err(stream) => stream,
     };
 
-    stream
-        .enumerate()
-        .map(|(index, res)| match serde_json::to_string(&res) {
-            Ok(bytes) => Ok(Bytes::from(bytes)),
-            Err(err) => {
-                // We didn't manage to serialise the response!
-                // Do our best to send some sort of error back.
-                serde_json::to_string(
-                    &graphql::FetchError::MalformedResponse {
-                        reason: err.to_string(),
-                    }
-                    .to_response(index == 0),
-                )
-                .map(Bytes::from)
+    let span = Span::current();
+    stream.enumerate().map(move |(index, res)| {
+        tracing::debug_span!(parent: &span, "serialize_response").in_scope(|| {
```

Comment:
:grey_question: shouldn't this be placed on the whole while-loop?

Reply:
that stream was there to send a serie of responses (with defer or stream), so it makes sense to have a separate `serialize_response` span for each. Right now we only a have a stream of one response.
I think that code should be rewritten instead, to handle defer elsewhere: https://github.com/apollographql/router/issues/299

Reply:
fair